sources/engine/Stride.Physics/Engine/PhysicsProcessor.cs

```diff
@@ -103,6 +103,12 @@ protected override bool IsAssociatedDataValid(Entity entity, PhysicsComponent ph
 
         protected override void OnEntityComponentAdding(Entity entity, PhysicsComponent component, AssociatedData data)
         {
+            if (currentFrameRemovals.Contains(component))
```

Comment:
Contains/Remove are O(n) so we better avoid this (O(n^2) if we add n physics component, since it's called for every new component).
Let's turn `currentFrameRemovals` into a `HashSet<>`.
Then you can also simplify this code:
```
if (currentFrameRemovals.Remove(component))
    return;
```